### DISCUSSION TITLE
http sink support body

### DISCUSSION THREAD

**Discussion by Radcriminal** [vectordotdev/vector#15126] [General]
Hey! 
I would like to know, does http sink support not only headers, but body as well? It's crucial for my task to send webhooks containing body.

**Comment by jszwedko** [comment]
Hi @Radcriminal ! Yes, events are encoded to create the HTTP request body according to the configured `encoding` and `framing` options.

**Reply by Radcriminal** [reply]
Thank you, but I experimented with this fields but did not get any result

I want to send message to slack. I have a webhook that waits for POST request with "text" field required. 
After some transforms I have a JSON object (verified by writing into file)

```json
{
    "msg":
    [
        {
        }
    ],
    "field1": "value1",
    "field2": "value2",
    "field_n": "value_n",
    "text": "My text that should be posted in slack"
}
```

Do I understand correctly, that message above would be sent in the body?

I have following configuration:
[sinks.to_slack]
type = "http"
inputs = [ "add_text" ]
uri = "https://hooks.slack.com/services/xxxxx/yyyyy/zzzzz"

I also experimented with encoder and framing delimiters. But I don't receive any message in Slack. Does vector has any debug option where I can see http response from slack?

P.S using curl everything works fine.

EDITED: I turned on Debug logs and see that request to slack not formed well, looks like body is empty with encoding text and delimiter "newline_delimited", but encoding json and newline_delimited worked well. I am a little bit confused and now wondering, what's the difference between encoding json and text?

**Reply by jszwedko** [reply]
Glad you were able to figure it out! The `text` encoding is a bit confusing. When used, it only encodes the value of the `message` key. This encoding primarily exists to support "pass-thru" configurations like `kafka` source to `kafka` sink where you just want the raw payload.
